javascript/javascript1/week3/note taking app/notetaking.js

```diff
@@ -0,0 +1,56 @@
+let notes = [];
+
+function saveNote(content, id) {
+  const note = {
+    content: content,
+    id: id,
+    completed: false,
+  };
+  notes.push(note);
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Cut vegetables", 3);
+saveNote("Clean the cupboard", 4);
+
+console.log(notes);
+
+function getNote(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (id === notes[i].id) {
+      return notes[i];
+    }
+  }
+  return `ID ${id} is not in the array`;
+}
+
+const firstNote = getNote(5);
+console.log(firstNote);
+
+function logOutNotesFormatted() {
+  for (let note of notes) {
+    console.log(
+      `The note with id: ${note.id}, has the following note text ${
+        note.content
+      } and it is ${note.completed ? "completed" : "not completed"}.`
+    );
+  }
+}
+
+logOutNotesFormatted();
+
+function markNoteAsCompleted(id) {
+  const note = notes.find((note) => note.id === id);
```

Comment:
Great work on the notes implementation! ⭐ Until this point you have been using a mix of old-school for loops and `for...of`, but here you've used `.find` instead. Why did you make this choice? 😄 

As this is also a somewhat advanced feature, I want to make sure you fully understand it! Can you explain what `find` and `=>` does? 🙂

Reply:
Thanks for this question. During our last class, we learned the differences between **find** and **filter**, and we also did some examples. So I wanted to try using **find** in the above function. 

find: It returns the first element that satisfies the given condition
=> : It is called as Arrow function, which is a simplest form of writing a traditional function. It is most commonly used in filter(), find() :)

Hope I have good understanding of this